File: wlr/OGUR/OGUR/Creatures/StatType.cs
using System.Collections.Generic;
namespace OGUR.Creatures
{
    public class StatType
    {
        public const string HEALTH = "Health";
        public const string MANA = "Mana";
        public const string STRENGTH = "Strength";
        public const string WISDOM = "Wisdom";
        public const string DEFENSE = "Defense";
        public const string LUCK = "Luck";
        public const string AGE = "Age";
        public const string HEIGHT = "Height";
        public const string WEIGHT = "Weight";
        public const string MOVE_COOL_DOWN = "MoveCoolDown";
        public const string PIETY = "Piety";

        public static readonly string[] Values = 
        {
            HEALTH, 
            MANA, 
            STRENGTH, 
            WISDOM, 
            DEFENSE, 
            LUCK, 
            AGE, 
            HEIGHT, 
            WEIGHT, 
            MOVE_COOL_DOWN, 
            PIETY
        };
    }
}

File: aigilas/src/aigilas/creatures/StatType.java
package aigilas.creatures;

public enum StatType {
    Health("H"),
    Energy("E"),
    Strength("S"),
    Wisdom("W"),
    Defense("D"),
    Luck("L"),
    Age("A"),
    Height("HE"),
    Weight("WE"),
    Move_Cool_Down("M"),
    Piety("P"),
    Regen("R");

    private String shortHand;

    private StatType(String shortHand) {
        this.shortHand = shortHand;
    }

    public static StatType get(String value) {
        for (StatType s : values()) {
            if (s.toString().equalsIgnoreCase(value)) {
                return s;
            }
        }
        return null;
    }

    public static StatType getShort(String value) {
        for (StatType s : values()) {
            if (s.shortHand.equalsIgnoreCase(value)) {
                return s;
            }
        }
        return null;
    }
}
